wordpress-mobile/WordPress-Android
# Site-creation rework: Site creation domain screen

Fixes #6928 

Known issue: the suggestions coming out of the FluxC domain suggestions api call don't seem to respect the wpcom-subdomains-only flag. This will be handled in a separate PR.

Implementation info:
* Similar to the headless "loader" Fragment in the themes selection screen, a headless fragment is employed here too (`SiteCreationDomainLoaderFragment`) for handling the communication with FluxC in a rotation-friendly manner, without losing the result of the network call if that happens to arrive during a device rotation.
* The whole UI is implemented as content of a RecyclerView so scrolling can be easy and has enough touch height to be performed when device height is small (think: landscape orientation). The textual header ("Step 4 of 4..") could be static and non-scrolling but in this first implementation I opted to keeping it in for maximum scrollable height.

To test:
* With the app logged in wpcom, create a new wpcom site via the site picker
* Reach the domain selection screen
* Notice the progress indicator whenever there is an active query in progress
* Enter a query text to trigger a new query and notice the list refreshing after a while
* Hit the "Create site" button to submit the selection. A Toast should pop up with the selected domain name.
